MVP component documentation for components A-D (#326)
primer/design: content/components/action-menu.mdx

<div>
<Link href="https://primer.style/react/ActionMenu">
<img alt="Action menu" src="https://user-images.githubusercontent.com/293280/123880964-95b3a700-d8f8-11eb-9775-cbaf165207ed.png" />
</Link>
</div>
<div>

<Heading sx={{fontSize: 3}}>To perform an action</Heading>
<Box as="p">

Action menus are used for disambiguation, navigation, or to display secondary options. They appear when users interact with buttons, actions, or other controls.

</Box>
</div>
<div>
<Link href="https://primer.style/react/ActionMenu#with-selection">
<img alt="Action Menu with selection" src="https://user-images.githubusercontent.com/293280/123881067-cdbaea00-d8f8-11eb-98e4-e57c64489308.png" />
</Link>
</div>
<div>
<Heading sx={{fontSize: 3}}>To make one or more selections</Heading>
<Box as="p">

Action menu can be used for making a selection among a small list of options. The list is usually predefined with system values, but in some cases it can include user-provided data when those are known not to grow into too many items.

</Box>
</div>
</Box>

## Related components
- [Action list](/design/components/action-list)

> - [Select panel](https://primer.style/react/SelectPanel)

Reviewer: Make sure we don't forget to update these links once we add the design docs.

---
title: Action menu
react: https://primer.style/react/ActionMenu
figma: https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?node-id=17091%3A66640

> description: Action menu is a component based on Action list for creating a menu of actions that expands through a trigger button.

Author: > This capitalization of "Action" looks strange: `...on Action list...`. I think we should use sentence case and make it a link. Something like "...based on the [action list](https://primer.style/design/components/action-list) component...".

Agreed. I can swap this so it's always sentence case. We should mention this in docs guidelines too.

> description: Action menu is a component based on Action list for creating a menu of actions that expands through a trigger button.

Reviewer 2: Agree with sentence case usage here, except at the start of a sentence.

Related: @langermank shared this ADR on component spelling: https://github.com/github/primer/blob/main/adrs/2022-02-15-spelling-of-component-names.md#:~:text=When%20referring%20to%20a%20UI%20pattern%20in%20general%20and%20not%20to%20a%20specific%20implementation%20(React%2C%20Rails%2C%20CSS%2C%20Figma)%20of%20a%20component%2C%20use%20sentence%20case%20(%22action%20list%22%2C%20or%20%22Action%20list%22%20if%20at%20the%20start%20of%20a%20sentence).

---

import {Box, Button, Heading, Link} from '@primer/react'

## Usage

> ### In overlays

Reviewer: Aren't action menus _always_ in an overlay?

Reviewer 3: I think so Mike, yes 🤔 it just _is_ an overlay menu

Reviewer: So maybe we don't need the "In overlays" heading?

<Box display="grid" gridTemplateColumns={['1fr', null, null, null, '1fr 2fr']} gridGap={4}>
<div>
<Link href="https://primer.style/react/ActionMenu">
<img alt="Action menu" src="https://user-images.githubusercontent.com/293280/123880964-95b3a700-d8f8-11eb-9775-cbaf165207ed.png" />
</Link>
</div>
<div>

> <Heading sx={{fontSize: 3}}>Action menu</Heading>

Reviewer: ```suggestion
<Heading sx={{fontSize: 3}}>To perform an action</Heading>
```

<Box as="p">

Action menus are used for disambiguation, navigation, or to display secondary options. They appear when users interact with buttons, actions, or other controls.

</Box>
</div>
<div>
<Link href="https://primer.style/react/ActionMenu#with-selection">
<img alt="Action Menu with selection" src="https://user-images.githubusercontent.com/293280/123881067-cdbaea00-d8f8-11eb-98e4-e57c64489308.png" />
</Link>
</div>
<div>

> <Heading sx={{fontSize: 3}}>Action menu with selection</Heading>

Reviewer: ```suggestion
<Heading sx={{fontSize: 3}}>To make one or more selections</Heading>
```

<Box as="p">

> ActionMenu can be used for making a single selection among a small list of options. The list is usually predefined with system values, but in some cases it can include user-provided data when those are known not to grow into too many items.

Reviewer: > for making a single selection

I think action menus can be used to make multiple selections too.

Author: whoops, pulled this directly from the docs in [action list here](https://primer.style/design/components/action-list#in-overlays). Will adjust
